ci/ray_ci/automation/determine_microcheck_tests.py

```diff
@@ -0,0 +1,79 @@
+import click
+from typing import Set, Dict
+
+from ci.ray_ci.utils import logger, ci_init
+from ray_release.configs.global_config import get_global_config
+from ray_release.test import Test
+from ray_release.result import ResultStatus
+
+LINUX_PYTHON_TEST_PREFIX = "linux:__python"
+
+
+@click.command()
+@click.argument("team", required=True, type=str)
+@click.argument("coverage", required=True, type=int)
+@click.option("--test-history-length", default=100, type=int)
+@click.option("--test-prefix", default=LINUX_PYTHON_TEST_PREFIX, type=str)
+def main(team: str, coverage: int, test_history_length: int, test_prefix: str) -> None:
+    """
+    This script determines the tests that need to be run to cover a certain percentage
+    of PR failures, based on historical data
+    """
+    ci_init()
+    tests = [
+        test for test in Test.gen_from_s3(test_prefix) if test.get_oncall() == team
+    ]
```

Comment:
how many tests is this? and how long time will it take this thing load?

and this is going to run on every PR commit? is there going to be cost concern?

Reply:
it takes about 10 minutes, we only need to generate the list of tests once a while, maybe nightly; there are 362 core tests